airflow/providers/microsoft/azure/CHANGELOG.rst

```diff
@@ -27,6 +27,31 @@
 Changelog
 ---------
 
+7.0.0
+.....
+
+Breaking changes
+~~~~~~~~~~~~~~~~
+
+.. warning::
+  In this version of the provider, we have changed AzureFileShareHook to use azure-storage-file-share API instead of azure-storage-file,
+  removed deprecated extra__azure_fileshare__ prefix from connection extras and removed protocol param from connection extras
+
+* get_conn from AzureFileShareHook return None instead FileService
+* Remove protocol param from Azure fileshare connection extras
+* Remove deprecated extra__azure_fileshare__ prefix from Azure fileshare connection extras
+* Remove share_name, directory_name param from AzureFileShareHook.check_for_directory in favor of AzureFileShareHook share_name and directory_path param
+* Remove share_name, directory_name param from AzureFileShareHook.check_for_file method in favor of AzureFileShareHook share_name and directory_path param
+* Remove share_name, directory_name param from AzureFileShareHook.list_directories_and_files
+* Remove share_name, directory_name param from AzureFileShareHook.list_files in favor of AzureFileShareHook share_name and directory_path param
+* AzureFileShareHook method create_share accept kwargs from ShareServiceClient.create_share instead of FileService.create_share
+* AzureFileShareHook method delete_share accept kwargs from ShareServiceClient.delete_share instead of FileService.delete_share
+* Remove share_name, directory_name param from AzureFileShareHook.create_directory in favor of AzureFileShareHook share_name and directory_path param
+* Remove share_name, directory_name, file_name param from AzureFileShareHook.get_file in favor of AzureFileShareHook share_name and file_path
+* Remove share_name, directory_name, file_name param from AzureFileShareHook.get_file_to_stream in favor of AzureFileShareHook share_name and file_path
+* Remove share_name, directory_name, file_name param from AzureFileShareHook.load_file in favor of AzureFileShareHook share_name and file_path
+* Remove AzureFileShareHook.load_string, AzureFileShareHook.load_stream in favor of AzureFileShareHook.load_data
```

Comment:
```suggestion
* Remove protocol param from Azure fileshare connection extras
* Remove deprecated extra__azure_fileshare__ prefix from Azure fileshare connection extras
* Remove share_name, directory_name param from the following in favor of AzureFileShareHook share_name and directory_path param
    * AzureFileShareHook.check_for_directory
    * AzureFileShareHook.check_for_file
    * AzureFileShareHook.list_files
    * AzureFileShareHook.create_directory
    * AzureFileShareHook.get_file
    * AzureFileShareHook.get_file_to_stream
    * AzureFileShareHook.load_file
* Remove share_name, directory_name param from AzureFileShareHook.list_directories_and_files
* AzureFileShareHook method create_share accept kwargs from ShareServiceClient.create_share instead of FileService.create_share
* AzureFileShareHook method delete_share accept kwargs from ShareServiceClient.delete_share instead of FileService.delete_share
* Remove AzureFileShareHook.load_string, AzureFileShareHook.load_stream in favor of AzureFileShareHook.load_data
```
Should we consolidate similar changes into one nested list?

Reply:
I have updated it a bit, PTAL